src/main/java/org/helpberkeley/memberdata/Restaurant.java

```diff
@@ -47,14 +48,13 @@ protected Restaurant(String name, int lineNumber) {
     }
 
     public static Restaurant createRestaurant(ControlBlock controlBlock, String name, int lineNumber) {
-        switch (controlBlock.getVersion()) {
-            case Constants.CONTROL_BLOCK_VERSION_200:
-                return new RestaurantV200(controlBlock, name, lineNumber);
-            case Constants.CONTROL_BLOCK_VERSION_300:
-                return new RestaurantV300(controlBlock, name, lineNumber);
-            default:
-                throw new MemberDataException("Control block version " + controlBlock.getVersion()
-                        + " is not supported for restaurant creation");
+        if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_200)) {
+            return new RestaurantV200(controlBlock, name, lineNumber);
+        } else if (controlBlock.versionIsCompatible(Constants.CONTROL_BLOCK_VERSION_300)) {
+            return new RestaurantV300(controlBlock, name, lineNumber);
+        } else {
+            throw new MemberDataException(MessageFormat.format(ControlBlock.UNSUPPORTED_VERSION_FOR,
+                    controlBlock.getVersion(), "restaurant creation."));
```

Comment:
My first comment was to add a constant for "restaurant creation", but when grepping for it, I didn't see it in any of the test cases.  So either we are missing coverage for this, or there is a test case that could be more specific in the error messages it is looking for.

Reply:
I've traced the code and I'm 99% sure this can never be hit as in all cases the version will already have been checked by this point. Should I still add a unit test for it?

Reply:
Leave it for the moment and next time we chat we can talk about ways to handle this.